tor/controller.go

```diff
@@ -360,48 +386,59 @@ func supportsV3(version string) error {
 	return nil
 }
 
+// The ProtocolInfo struct is used to wrap everything we get in the response to PROTOCOLINFO
+// into a neat package, so we can pass it around as a single object.
+type ProtocolInfo struct {
```

Comment:
doesn't need to be exported

Reply:
What do I need to do? I don't know go at all.

Reply:
In go a capital letter indicates that something is exported, ie can be called from outside of the package. A lower case letter reserves it only for use within the package, which would be suitable in this case because we don't need it anywhere else. 

[Effective go](https://golang.org/doc/effective_go.html) is a good read for getting to know golang if you want to delve deeper :)